lib/new_relic/telemetry_sdk/buffer.rb

```diff
@@ -0,0 +1,50 @@
+# encoding: utf-8
+# frozen_string_literal: true
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-telemetry-sdk-ruby/blob/main/LICENSE for complete details.
+
+module NewRelic
+  module TelemetrySdk
+    class Buffer
+      attr_reader :items
+      attr_accessor :common_attributes
+
+      CAPACITY = 2000
+
+      def initialize common_attributes=nil
+        @items = []
+        @common_attributes = common_attributes
+        @lock = Mutex.new
+        @capacity = capacity
+      end
+
+      # Items recorded into the buffer must have a to_h method for transformation
+      def record item
+        @lock.synchronize do
+          if size < capacity
+            @items << item
+          end
+        end
+      end
+
+      def flush
+        data = nil
+        @lock.synchronize do
+          data = @items.map(&:to_h)
+          @items = []
+        end
+        return data, @common_attributes
+      end
+
+      alias_method :to_h, :flush
+
+      def size
+        @items.length
+      end
+
+      def capacity
+        CAPACITY
```

Comment:
Are we going to eventually allow user to configure `CAPACITY`?  If not, probably should just refer to the `CAPACITY` directly.  Right now, we have `CAPACITY`, `@capacity` AND `def capacity`, all presumably the same thing and unchanging at runtime.

Reply:
Oops, I didn't mean to have `@capacity` still there since it is no longer used. Thanks for catching that.

Currently there is no plan to make capacity configurable. The hard limit of 2000 items is a MUST [in the spec](https://github.com/newrelic/newrelic-telemetry-sdk-specs/blob/master/limits.md#data-point-limit). The reason why I wrote a method solely to return the constant is because it makes it easier to test. Definitely open to other strategies there, though.